pkg/controller_helper/model_source/modelsource.go

```diff
@@ -59,12 +60,28 @@ type ModelSourceProvider interface {
 
 func NewModelSourceProvider(model *coreapi.OpenModel) ModelSourceProvider {
 	if model.Spec.Source.ModelHub != nil {
+
+		// fileName should not in ignorePatterns
+		if model.Spec.Source.ModelHub.Filename != nil && model.Spec.Source.ModelHub.IgnorePatterns != nil {
```

Comment:
can we verify this in webhooks? As I mentioned, once filename is set, the other patterns should not be configured anyway. It's useless.